client/components/data/page-templates-data/index.jsx

```diff
@@ -49,6 +51,10 @@ export default React.createClass( {
 
 	componentWillReceiveProps( nextProps ) {
 		if ( nextProps.siteId === this.props.siteId ) {
+			if ( this.props.currentTheme && this.props.currentTheme.stylesheet !== this.activeTheme ) {
```

Comment:
This should be checking `nextProps` I think, to compare the _incoming_ theme against the currently tracked active theme.